src/test/resources/com/puppycrawl/tools/checkstyle/checks/coding/superfinalize/InputSuperFinalizeMethodReference.java

```diff
@@ -0,0 +1,19 @@
+package com.puppycrawl.tools.checkstyle.checks.coding.superfinalize;
+
+public class InputSuperFinalizeMethodReference extends ClassWithFinalizer {
+
+    public interface CheckedConsumer<E extends Throwable> {
+        void get() throws E;
+    }
+
+    @Override
+    protected void finalize() throws Throwable {
+        CheckedConsumer<Throwable> r = super::finalize;
+        r.get();
+    }
+}
+
+class ClassWithFinalizer {
+    protected void finalize() throws Throwable {
```

Comment:
please add `//violation` comment and comment over class for Check config.

One day in future we will read config and expected violations from Input file.

Reply:
done